consensus/mempool_test.go

```diff
@@ -202,76 +207,3 @@ func TestMempoolRmBadTx(t *testing.T) {
 		return
 	}
 }
-
-// CounterApplication that maintains a mempool state and resets it upon commit
-type CounterApplication struct {
```

Comment:
Why do we need to remove this Application?
Are we sure all the aspects tested by unit tests using it are still covered?

Reply:
This was done in v0.36 as part of an effort to reduce the amount of abci implementations we had to make it easier to maintain